### PR TITLE
Add STARSolo

### PR DESCRIPTION
**Pull request by mtekman**
FOR CONTRIBUTOR:
* [X] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [X] - License permits unrestricted use (educational + commercial)
* [X] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [X] - This PR does something else (explain below)

This is a wrapper of the STARSolo capabilities of the new version of RNA STAR which provides a single-click solution to 10X datasets following the CellRanger model. 

This PR is forked off the much-needed [facelift PR](https://github.com/galaxyproject/tools-iuc/pull/2359/) given by @wm75

### CODE REVIEW

**Comment by bgruening** [comment]
> AssertionError: Test file (737K-august-2016.txt) is missing. If you use planemo try --update_test_data to generate one.

Maybe a rebase is needed as well?

File: tools/rgrnastar/rg_rnaStarSolo.xml [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+                    </param>
+                    <when value="with-gtf">
+                        <expand macro="@SJDBOPTIONS@" optional="false"/>
+                    </when>
+                    <when value="without-gtf" />
+                </conditional>
+            </when>
+        </conditional>
+
+        <section name="solo" title="Star Solo Settings" expanded="true">
```
Comment (bgruening) [inline tools/rgrnastar/rg_rnaStarSolo.xml]: its a little bit strange that the tool is called start solo but only here the "star solo settings" begin ... I would assume all settings are star solo settings :)

Comment (wm75) [reply tools/rgrnastar/rg_rnaStarSolo.xml]: Yes, the idea was to get the tool working, then determine the extent of overlap with STAR and split all that out into macros.

**Comment by mtekman** [comment]
I did not add an `auto_tool_repositories` or a `suite` to the shed because STARSolo is still very much a STAR tool, and derives all of its components from it, so I thought it would be odd to split these into separate toolshed repos.

**Comment by bgruening** [comment]
During GCC we decided to split them always into separate repos, more for technical reasons. Is this ready for review?

**Comment by mtekman** [comment]
I think so, I added the `auto_tool_repositories` based on the mothur model, but perhaps this is overkill, since there are only two wrappers need here?

File: tools/rgrnastar/.shed.yml [hunk 2]
```diff
@@ -1,15 +1,31 @@
...
-  automated dependency installation by Ross Lazarus.
+  Most of this wrapper was originally written by Jeremy Goecks, with kibbitzing and automated dependency installation by Ross Lazarus.
+
+  The scRNA-seq mapping and quantification component is encapsulated within the STARSolo wrapper.
 homepage_url: https://github.com/alexdobin/STAR
 name: rgrnastar
 owner: iuc
 remote_repository_url: https://github.com/galaxyproject/tools-iuc/tree/master/tools/rgrnastar
 type: unrestricted
+auto_tools_repositories:
```
Comment (bgruening) [inline tools/rgrnastar/.shed.yml]: The auto one, unfortunately, will not work. The tool id is `rna_star` but the repro was called `rgrnastar`.

I think you need to use something like this: https://github.com/galaxyproject/tools-iuc/pull/2517/files#diff-05cc38f519e7fe8f2effa3958a835c32R10

Comment (mtekman) [reply tools/rgrnastar/.shed.yml]: ah good example, I will adopt this

Comment (mtekman) [reply tools/rgrnastar/.shed.yml]: @bgruening I changed the shed.yml, would this work?

**Comment by bgruening** [comment]
I uploaded the tools to the test-toolshed and it looks good to me.
@wm75 ready when you are.

**Comment by bgruening** [comment]
I updated Star to the latest version which was released yesterday.

**Pull request merged**